packages/graph/hash_graph/lib/graph/src/api/rest/link.rs

```diff
@@ -31,7 +34,8 @@ use crate::{
             CreateLinkRequest,
             RemoveLinkRequest,
             KnowledgeGraphQuery,
-            LinkRootedSubgraph
+            LinkRootedSubgraph,
+            PersistedLinkMetadata
```

Comment:
Do we not want to use this anywhere? Links are a bit buggered so I understand if we don't.

Reply:
But I assumed we still need to make the type available here so that it is in the generated `api.ts` file

Reply:
I think for consistency we'd want to return the metadata from the `create` and `update` methods though, no?

Reply:
Probably, but did not think that this PR should be the place to do it as I presumed there was a reason for the temporary inconsistency. But perhaps there wasn't.

@TimDiekmann what was the reason for returning the whole `PersistedLink` for these methods, rather than an "identifier" struct similar to the rest of the API?

Reply:
We return `PersistedLink` in `get_entity_links`

Reply:
Because `PersistedLink` _was_ basically the identifier. That's one of the problems with the current link setup, we don't have the concept of a LinkId because we were uniquely identifying them by their Source, Link Type, Destination.

We're already adjusting all of the return types of the methods, I think this PR is the correct place to do it :)

Reply:
As you pointed out offline, moving the identifier information into the metadata would either result in a lot of duplication with `PersistedLink` (like I said, it basically is the identifier), or it would leave `Link` with a single field within it.

Both of those seem like terrible options, and this is caused by us not having an appropriate identifier for links, which we know we need to fix soon by introducing an ID rather than this triplet of fields (which is incorrect anyway).

As such, I think it's fine to leave it as it is, despite being a bit inconsistent. Fixing it would be way out of scope of this PR.